### ISSUE TITLE
Unexpected behavior with region and nextgroup in syntax files

### ISSUE THREAD

**Issue by tmccombs** [vim/vim#6378]
**Describe the bug**
usr_44.txt states:

> The main difference is that a match
item is a single pattern, which must match as a whole.  A region starts as
soon as the "start" pattern matches.  Whether the "end" pattern is found or
not doesn't matter.  Thus when the item depends on the "end" pattern to match,
you cannot use a region.

However, when using a region as the `nextgroup` for another rule, it doesn't seem to match unless the end pattern is present.

**To Reproduce**
1. Create a syntax file containing the following:
```
syn match myStart "start" nextgroup=myString skipwhite
syn region myString start=+"+ end=+"+ oneline contained

hi def link myStart Label
hi def link myString String
```

2. In  a new buffer source the syntax file, and enter the text `start "a`. The `"a` text is not highlighted as a string
3. Type a closing `"`, the quoted string is now highlighted.

**Expected behavior**
The text should start highlighting as a string as soon as the start pattern matches. It should not need to wait for the end pattern.

**Screenshots**
![image](https://user-images.githubusercontent.com/2541726/86330558-11efad80-bc05-11ea-95ec-2b73b6a67cf7.png)
![image](https://user-images.githubusercontent.com/2541726/86330596-1f0c9c80-bc05-11ea-819c-00f520899e9b.png)

**Environment (please complete the following information):**
 - Vim version 8.0.1453 (also in neovim 0.4.3)
 - OS: Ubuntu 18.04
 - Terminal: alacritty

**Comment by ddickstein** [comment]
I don't think this is related to `nextgroup` being a region. I think this is due to the `oneline` keyword. If you remove that from your region definition you will get highlighting. From the `oneline` help:

> It must match completely in the current line...the "end" pattern must still match in the first line, otherwise the region doesn't even start.

I take this to mean that the end pattern needs to actually be present in the line before the match takes effect.

**Comment by tmccombs** [comment]
hmm, so how would you define a string like that that must occur on a single line?

**Comment by ddickstein** [comment]
I don't actually know what the main use cases for the `online` keyword are, so someone more familiar will need to weigh in there. But if your goal is to make the string highlighting stop either at the end of the line or at the terminated string, I think you can just use `end=+\("\|$\)+`
